KarhooSDK/Api/DataStore/User/UserDataStore.swift

```diff
@@ -76,7 +76,17 @@ final class DefaultUserDataStore: UserDataStore {
            let expiryTimeInterval = TimeInterval(expiryDateString) {
             expiryDate = Date(timeIntervalSince1970: expiryTimeInterval)
         }
-        return Credentials(accessToken: accessToken, expiryDate: expiryDate, refreshToken: refreshToken)
+        var refreshTokenExpiryDate: Date?
+        if let refreshTokenExpiryDateString = secretStore.readSecret(withKey: CredentialsStoreKeys.refreshToken.rawValue),
```

Comment:
I would rename this to `refreshTokenExpiryIntervalString`. I find it a bit misleading as it sounds like we're storing the end calculated date already.